### ISSUE TITLE
Nested notation not printed

### ISSUE THREAD

**Issue by samuelgruetter** [coq/coq#6304]
```
Notation "'for' m 'from' 0 'to' N 'updating' ( s1 ) {{ b }} ;; rest" :=
  (let s1 :=
    (fix rec(n: nat) := match n with
     | 0 => s1
     | S m => let s1 := rec m in b
     end) N
  in rest)
  (at level 20).

Definition multiply(a b: nat): nat :=
  let res := 0 in
  for i from 0 to a updating (res) {{
    for j from 0 to b updating (res) {{ S res }};;
    res
  }};; res.

Print multiply.
```

prints the following:

```
multiply = 
fun a b : nat =>
let res := 0 in
let res0 :=
  (fix rec (n : nat) : nat :=
     match n with
     | 0 => res
     | S i =>
         let res0 := rec i in
         for j from 0 to b updating (res0){{S res0}};; res0
     end) a in
res0
     : nat -> nat -> nat
```
That is, for the inner for loop, it prints the notation, but for the outer for loop, it does not.

I would like the notation to be printed also for the outer loop.

Version: 8.6

**Comment by herbelin** [comment]
This is fixed in 8.7.0. Unfortunately, I don't know which exact commit fixed it (speculatively dfdaf4d or 4e70791). Backporting it to 8.6 would need some time.

**Comment by samuelgruetter** [comment]
Just installed 8.7.0, and it indeed prints the notations the way it should, problem solved :smiley: 
Thanks for the quick reply!
